servers/visual_server.cpp

```diff
@@ -1429,8 +1429,10 @@ Array VisualServer::_get_array_from_surface(uint32_t p_format, PoolVector<uint8_
 						PoolVector<Vector3>::Write w = arr_3d.write();
 
 						for (int j = 0; j < p_vertex_len; j++) {
-							const float *v = (const float *)&r[j * total_elem_size + offsets[i]];
-							w[j] = Vector3(v[0], v[1], v[2]);
+							const float &x = r[j * total_elem_size + offsets[i]];
+							const float &y = r[j * total_elem_size + offsets[i] + 1];
+							const float &z = r[j * total_elem_size + offsets[i] + 2];
```

Comment:
Is this correct given that r seems to be 8 bit?
```
PoolVector<uint8_t>::Read r = p_vertex_data.read();
```
Use brackets for precedence to make it clear, but I think this does e.g.
```
const float &x = r[(j * total_elem_size) + offsets[i] + 1];
```
The +1 is 1 byte, therefore unaligned.

Reply:
This doesn't seem correct?    Since r is a byte buffer, wouldn't you want to index:

```
const float &y = r[j * total_elem_size + offsets[i] + 4]
const float &z = r[j * total_elem_size + offsets[i] + 8]
```